### PR TITLE
Widget default size feature was added

### PR DESCRIPTION
**Pull request by brutskov**

### CODE REVIEW

File: client/app/_dashboards/dashboard.controller.js [hunk 1]
```diff
@@ -278,6 +278,18 @@ const dashboardController = function dashboardController(
...
+    function resolveWidgetLocation(widget) {
+        let widgetLocationStr = angular.copy(defaultWidgetLocation);
+        if (widget.defaultSize) {
+            let widgetLocation = jsonSafeParse(widgetLocationStr);
+            widgetLocation.height = widget.defaultSize.height;
+            widgetLocation.width = widget.defaultSize.width;
+
+            widgetLocationStr = jsonSafeStringify(widgetLocation);
+        }
+        return  getNextEmptyGridAreaV2(widgetLocationStr);
```
Comment (DmitryMishin) [inline client/app/_dashboards/dashboard.controller.js]: 2 spaces

File: client/app/components/modals/widget-wizard/widget-wizard.controller.js [hunk 2]
```diff
@@ -353,8 +360,9 @@ const widgetWizardController = function WidgetWizardController($scope, $mdDialog
                     }
                 } else if(widget.widgetTemplate.type === 'TABLE') {
                     columns = JSON.parse(widget.widgetTemplate.chartConfig).columns;
+                    defaultSize = JSON.parse(widget.widgetTemplate.chartConfig).defaultSize;
```
Comment (DmitryMishin) [inline client/app/components/modals/widget-wizard/widget-wizard.controller.js]: what if a widget is not a table?

File: client/vendors/ngecharts.js [hunk 3]
```diff
@@ -17,7 +17,8 @@
                 withLegend: '=withLegend',
                 forceWatch: '=forceWatch',
                 chartActions: '=chartActions',
-                config: '=config'
+                config: '=config',
+                configCallback: '&configCallback'
```
Comment (DmitryMishin) [inline client/vendors/ngecharts.js]: add comma

File: client/vendors/ngecharts.js [hunk 4]
```diff
@@ -47,6 +48,10 @@
                     opts = JSON.parse(opts);
                 }
 
+                if (scope.configCallback) {
+                    scope.configCallback({arg1: opts});
+                }
```
Comment (DmitryMishin) [inline client/vendors/ngecharts.js]: add spaces before/after brackets

**Review by dzmitrymishyn — CHANGES_REQUESTED** [review]

File: client/app/_dashboards/dashboard.controller.js [hunk 5]
```diff
@@ -278,6 +278,18 @@ const dashboardController = function dashboardController(
         });
     };
 
+    function resolveWidgetLocation(widget) {
+        let widgetLocationStr = angular.copy(defaultWidgetLocation);
+        if (widget.defaultSize) {
+            let widgetLocation = jsonSafeParse(widgetLocationStr);
+            widgetLocation.height = widget.defaultSize.height;
+            widgetLocation.width = widget.defaultSize.width;
+
```
Comment (danymill) [inline client/app/_dashboards/dashboard.controller.js]: redundant empty line

File: client/app/_dashboards/dashboard.controller.js [hunk 6]
```diff
@@ -278,6 +278,18 @@ const dashboardController = function dashboardController(
         });
     };
 
+    function resolveWidgetLocation(widget) {
+        let widgetLocationStr = angular.copy(defaultWidgetLocation);
```
Comment (danymill) [inline client/app/_dashboards/dashboard.controller.js]: you don't need to clone string.

File: client/app/components/modals/widget-wizard/widget-wizard.controller.js (hunk 2 above)
Comment (danymill) [inline client/app/components/modals/widget-wizard/widget-wizard.controller.js]: Please parse config once and then work with parsed data

File: client/vendors/ngecharts.js (hunk 3 above)
Comment (danymill) [inline client/vendors/ngecharts.js]: If you don't need to change binding names, remove they duplications

**Review by danymill — CHANGES_REQUESTED** [review]

**Review by dzmitrymishyn — APPROVED** [review]

**Review by danymill — APPROVED** [review]

**Pull request merged**